src/FakeItEasy/CallOrderingContext.cs

```diff
@@ -0,0 +1,99 @@
+namespace FakeItEasy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using FakeItEasy.Core;
+
+    /// <summary>
+    /// Represents a context to make assertions on the order of the calls made on a fake.
+    /// </summary>
+    public class CallOrderingContext : IDisposable

```

Comment:
Now that I think of it... we could make a `ICallOrderingContext` interface (as I initially did) and put the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute on the `CheckNextCall` method. This way, `IHideObjectMembers` will work as expected. Also, it's better to expose interfaces rather than concrete classes where possible.